### PR TITLE
fix(index): ensure to nullify `inputSourceMap` if `sourcemap` is unavailable (`devtool`)

### LINKED ISSUE #111: [v1.0.0-beta.2] sourcemap not work in parallel
first set cache true, parallel false
then parallel true, the sourcemap not work

### PR DESCRIPTION
**Pull request by evilebottnawi**
### `Notable Changes`

- Simplify logic based on https://github.com/webpack/webpack-sources/blob/91862553dc108e897ec24ece7b5bd8aae42620e5/lib/Source.js#L24
- Add tests

### `Issues`

- Fixes #111

### CODE REVIEW

File: test/__snapshots__/source-map-options.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,551 @@
...
+    "names": Array [
+      "module",
+      "exports",
+      "console",
+      "log",
+      "b",
+    ],
+    "sources": Array [
+      "/home/evilebottnawi/IdeaProjects/uglifyjs-webpack-plugin/test/fixtures/entry.js",
+    ],
```
Comment (d3viant0ne) [inline test/__snapshots__/source-map-options.test.js.snap]: This will always fail somewhere else with an absolute file path

Comment (evilebottnawi) [reply test/__snapshots__/source-map-options.test.js.snap]: @d3viant0ne fixed

File: src/index.js [hunk 2]
```diff
@@ -123,18 +123,14 @@ class UglifyJsPlugin {
...
-                if (asset.sourceAndMap) {
-                  const sourceAndMap = asset.sourceAndMap();
-                  inputSourceMap = sourceAndMap.map;
-                  input = sourceAndMap.source;
-                } else {
-                  inputSourceMap = asset.map();
-                  input = asset.source();
-                }
+              if (this.options.sourceMap && asset.sourceAndMap) {
+                const sourceAndMap = asset.sourceAndMap();
```
Comment (michael-ciniawsky) [inline src/index.js]:
```js
const { source, map } = asset.sourceAndMap();

input = source;
inputSourceMap = map;

sourceMap = new SourceMapConsumer(inputSourceMap)
```

File: src/index.js [hunk 3]
```diff
@@ -123,18 +123,14 @@ class UglifyJsPlugin {
...
+                input = sourceAndMap.source;
+                inputSourceMap = sourceAndMap.map;
                 sourceMap = new SourceMapConsumer(inputSourceMap);
               } else {
                 input = asset.source();
+                inputSourceMap = null;
```
Comment (michael-ciniawsky) [inline src/index.js]: This is the windows related fix ?

File: src/index.js (hunk 2 above)
Comment (michael-ciniawsky) [reply src/index.js]: Does `webpack >= v2.0.0` already support this convenience method ? So the `else` branch  **is** really legacy

File: src/index.js (hunk 3 above)
Comment (evilebottnawi) [reply src/index.js]: @michael-ciniawsky no, when `devtool` set to null we don't have `map`

File: src/index.js (hunk 2 above)
Comment (evilebottnawi) [reply src/index.js]: @michael-ciniawsky need investigate

File: src/index.js (hunk 3 above)
Comment (michael-ciniawsky) [reply src/index.js]: @evilebottnawi Sure, but this is nevertheless the fix for the parallel source maps bug :) ? It doesn't seem related to `options.parallel` exclusively then, should we reframe the commit message ?

Comment (evilebottnawi) [reply src/index.js]: @michael-ciniawsky in this commit i just simplify logic and add tests to make sure that everything works and the problem is not in the plugin

File: src/index.js (hunk 2 above)
Comment (michael-ciniawsky) [reply src/index.js]: Should be safe https://github.com/webpack/webpack/blob/webpack-2/package.json#L26

**Comment by michael-ciniawsky** [comment]
@evilebottnawi Status 😛 ?  Could you address the style nit, so we can get this in or is anything blocking this PR ?

**Comment by evilebottnawi** [comment]
@michael-ciniawsky done :+1:

**Pull request merged**